roombapy/roomba.py

```diff
@@ -158,10 +158,11 @@ def periodic_connection(self) -> None:
         # only one connection thread at a time!
         if self.periodic_connection_running:
             return
-        self.periodic_connection_running = True
         while not self.stop_connection:
             try:
-                self._connect()
+                if not self.periodic_connection_running:
+                    self._connect()
+                    self.periodic_connection_running = True
```

Comment:
Are you sure this does what you think it does? This looks like a mutex to me and with this you're defeating the purpose of the mutex?

Reply:
self.periodic_connection_running is just the current state of whether or not we have a current active connection.

self.stop_connection is the mutex-like that actually controls whether or not the connection SHOULD be stopped.

If stop_connection == False and periodic_connection_running == False, we need to restart the connection- anything else we just let pass through, because we're either already connected and should be, or stopping the connection and falling out of the while loop.

Reply:
Hmmm it doesn't look like this. `self.periodic_connection_running` is just prohibiting multiple `while`-loops from running. With your change it will be possible to start it twice.